pymc/distributions/continuous.py

```diff
@@ -777,11 +777,13 @@ def logp(
             norm = 0.0
 
         logp = _logprob(normal, (value,), None, None, None, mu, sigma) - norm
+        logp = at.switch(
```

Comment:
Actually to be equivalent to what we had before we should do something like this: 

https://github.com/pymc-devs/pymc/blob/c53cd2f5653f46c59206c66acd1fc1bc0ebf9e77/pymc/distributions/truncated.py#L316-L320

Reply:
Isn't that equivalent to what is implemented here because of the default values for lower and upper?
https://github.com/pymc-devs/pymc/blob/main/pymc/distributions/continuous.py#L714-L715

Reply:
We retrieve the None case here: https://github.com/pymc-devs/pymc/blob/91cbebda4890b36ceae12855ec35e59312f38de7/pymc/distributions/continuous.py#L763-L764

So in those cases we avoid introducing the useless switch. It's a small optimization but I don't see any reason yo modify it.

Reply:
Oh I see. OK

Reply:
The last commit makes the implementation here more analogous to the general truncated case you linked above - thanks for that pointer!

Reply:
Thanks @ricardoV94 for taking a look and helping out already! Let me know if the implementation in the latest few commits looks ok. Also, it looks like this PR is waiting for approval to run the full test suite.